HIP-xxxx.md

```diff
@@ -0,0 +1,163 @@
+# HIP-xxxx : Standard for TXT records
+
+```
+Number:  HIP-xxxx
+Title:   Standard for TXT records
+Type:    Standards
+Status:  Draft
+Authors: 0xStefan <https://twitter.com/0xStefan>
+Created: 2022-02-22
+```
+
+## Abstract
+
+A proposal for TXT record standards to provide on-chain data for a domain.
+
+## Motivation
+
+#### Domain Parking
+
+There's currently no way to know if a domain is for sale by the owner or not.
+A domain seller has no reliable way to provide that information.
+
+#### Identity / Web3 Profile
+
+Websites using Sign in with Handshake can use and render the TXT record data, e.g. as profile information. The domain owner is still in control of the data.
+
+## TXT Standards Definition
+
+For the above to work we need standards so any website can properly render the TXT records. The records should be flexible, short and easy to understand. Since it's impossible to validate the data via HSD. Therefore TXT records should be provided mostly without any prefix like "mailto:", "tel:" or "https://www.twitter.com/".
```

Comment:
I don't agree with this part. IMO, the protocol should be mandatory.

Some of them can be assumed based on the content. Eg: foo@bar.com is most likely a `mailto:`.
But there are no ways to determine if `http:` or `https:`. 
Also, it creates the possibility of different interpretation between readers/consumers.

parking.sinpapeles has this requirement and it is already used by a few thousands TLDs.
Not saying that PS should become the standard.

Reply:
hm. Yes, I agree that there is no way to know if it should be http or https if it's a URL.
So for URL's I'm totally with you. It should be mandatory to provide a fully qualified URL. 
I will revise this section.

For all the other data I still think it should not be included.

Let's look at some examples:

`github=Falci` is much cleaner and easier to handle than `github=https://github.com/Falci`
What if GitHub decides to change the link to https://github.com/profile/Falci? For site owners/apps this change would be really easy. For domain owners a real pain to change all of their domains.

`email=allison@bob.hns` 
`phone=+1 (536) 888 6253`
With the term "email" or "phone" it's already clear what the data will be. We should let the site owner or app decide how to render it. Also for domain owners it's also easier to understand and to handle.

Reply:
I agree about usernames, there's no need of full URLs for github and social links. And with @Falci about requiring protocol schemes for all fields with URLs (avatar, etc.)

>With the term "email" or "phone" it's already clear what the data will be. We should let the site owner or app decide how to render it.

Agreed, and to be able to render correctly, there needs to be some restriction / format for `email` and `phone`. I'm sure there are RFCs that define regexes for these. Phone might be a bit harder - one solution is WhatsApp's format (no +, include countrycode, only digits).
This way, apps can decide how to render them (prettify them with spaces and parantheses, etc.)

Reply:
> there needs to be some restriction / format for email and phone. I'm sure there are RFCs that define regexes for these.

I bet. If not we can provide it for each field. In the end the site or app that renders the data will always have to verify and clean the data before output. There is simply no other way. But we can make it easy for them by providing templates.

 > Phone might be a bit harder

If you want to use it as a link it's actually pretty easy. Those are all valid:

```
<a href="tel:+1(536)8886253">...
<a href="tel:+15368886253">...
<a href="tel:+1536-888-6253">...
<a href="tel:1536-888-6253">... (missing the "+" but most browsers will still get it)
```
You would just need to remove the spaces since they are not allowed according to [RFC 3966](https://www.rfc-editor.org/rfc/rfc3966)

Reply:
I updated the PR without the tel: and mailto:

My argument is that users are providing a phone number/email address in their profile data.
They are not providing a link to an email address nor a link to a phone number.

Site owners/apps should decide how they want to use and render that data.